src/treeview/filler.js

```diff
@@ -142,3 +144,33 @@ export function isBlockFiller( domNode, blockFiller ) {
 
 	return domNode.isEqualNode( templateBlockFiller );
 }
+
+/**
+ * Assign key observer which move cursor from the end of the inline filler to the begging of it when
+ * the left arrow is pressed, so the filler does not break navigation.
+ *
+ * @param {engine.treeView.TreeView} treeView TreeView instance we should inject quirks handling on.
+ */
+export function injectQuirksHandling( treeView ) {
+	treeView.on( 'keydown', jumpOverInlineFiller );
```

Comment:
For the future reference – it's fine to have this listener attached through `on()` as this code will be used by the tree view itself. Destruction of the tree view will ensure that no events are fired on it.